gen_examples.py

```diff
@@ -0,0 +1,244 @@
+
+class PeerConnection:
+  SESSION_SDP = \
+ """v=0
+    o=- {0.session_id} {0.version} IN IP4 0.0.0.0
+    s=-
+    t=0 0
+    a=ice-options:trickle
+    a=group:BUNDLE a1 v1
+    """
+# TODO: allow configurable BUNDLE
+
+  TRANSPORT_SDP_PRE_RTCP = \
+ """a=ice-ufrag:{0[ice_ufrag]}
+    a=ice-pwd:{0[ice_pwd]}
+    a=fingerprint:sha-256 {0[dtls_fingerprint]}
+    a=setup:{0[dtls_dir]}
+    a=dtls-id:1
+    """
+  TRANSPORT_SDP_RTCP = \
+ """a=rtcp:{0[local_rtcp]} IN IP4 {0[local_ip]}
+    """
+  TRANSPORT_SDP_POST_RTCP = \
+ """a=rtcp-mux
+    a=rtcp-rsize
+    """
+  TRANSPORT_SDP_WITH_RTCP = TRANSPORT_SDP_PRE_RTCP + TRANSPORT_SDP_RTCP + \
+                            TRANSPORT_SDP_POST_RTCP
+  TRANSPORT_SDP_NO_RTCP = TRANSPORT_SDP_PRE_RTCP + TRANSPORT_SDP_POST_RTCP
+# TODO: rtcp-mux-only
+
+  AUDIO_SDP = \
+ """m=audio {0[local_port]} UDP/TLS/RTP/SAVPF 96 0 8 97 98
+    c=IN IP4 {0[local_ip]}
+    a=mid:{0[mid]}
+    a=msid:{0[msid]}
+    a=sendrecv
+    a=rtpmap:96 opus/48000/2
+    a=rtpmap:0 PCMU/8000
+    a=rtpmap:8 PCMA/8000
+    a=rtpmap:97 telephone-event/8000
+    a=rtpmap:98 telephone-event/48000
+    a=maxptime:120
+    a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
+    a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
+    """
+  # TODO: proper default candidate in c=, m=, rtcp lines
+
+  VIDEO_SDP = \
+ """m=video {0[local_port]} UDP/TLS/RTP/SAVPF 100 101
+    c=IN IP4 {0[local_ip]}
+    a=mid:{0[mid]}
+    a=msid:{0[msid]}
+    a=sendrecv
+    a=rtpmap:100 VP8/90000
+    a=rtpmap:101 rtx/90000
+    a=fmtp:101 apt=100
+    a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
+    a=rtcp-fb:100 ccm fir
+    a=rtcp-fb:100 nack
+    a=rtcp-fb:100 nack pli
+    """
+
+  DATA_SDP = \
+ """m=application {0[local_port]} UDP/DTLS/SCTP webrtc-datachannel
+    c=IN IP4 {0[local_ip]}
+    a=mid:{0[mid]}
+    a=fmtp:webrtc-datachannel max-message-size=65536
```

Comment:
According to https://tools.ietf.org/html/draft-ietf-mmusic-sctp-sdp-21#section-6 there is no fmtp attribute any more. It should be 'a=max-message-size:65536' instead.